pycytominer/cyto_utils/DeepProfiler_processing.py

```diff
@@ -0,0 +1,160 @@
+"""
+Utility function to load and process the output files of a DeepProfiler run.
+"""
+import os
+import pathlib
+import numpy as np
+import pandas as pd
+
+from pycytominer import aggregate
+from pycytominer.cyto_utils import infer_cp_features, load_npz
+
+
+class AggregateDeepProfiler:
+    """This class holds all functions needed to load and annotate the DeepProfiler run.
+
+    If the class has public attributes, they may be documented here
+    in an ``Attributes`` section and follow the same formatting as a
+    function's ``Args`` section. Alternatively, attributes may be documented
+    inline with the attribute's declaration (see __init__ method below).
+
+    Properties created with the ``@property`` decorator should be documented
+    in the property's getter method.
+
+    Attributes
+    ----------
+    attr1 : str
+        Description of `attr1`.
+    attr2 : :obj:`int`, optional
+        Description of `attr2`.
+
+    """
+    def __init__(
+        self,
+        index_file,
+        profile_dir,
+        aggregate_operation="median",
+        aggregate_on="well",
+        file_delimiter="_",
+        file_extension=".npz",
+    ):
+        self.index_file = index_file
+        self.profile_dir = profile_dir
+        self.aggregate_operation = aggregate_operation
+        self.aggregate_on = aggregate_on
+        self.file_delimiter = file_delimiter
+        self.file_extension = file_extension
+        if not self.file_extension.startswith("."):
+            self.file_extension = f".{self.file_extension}"
+        self.index_df = pd.read_csv(index_file)
+
+    def build_filenames(self):
+        """
+        Single cell profile file names indicated by plate, well, and site information
+        """
```

Comment:
I am not sure how the docstrings should look like for functions with no parameters and also don't return anything. Should it just be the following?

```
Returns
--------
None
        Nothing is returned.
```

If that is the case then both `setup_aggregate` and `aggregate_deep` should also have the same docstrings.